warehouse/internal/model/validation.go

```diff
@@ -4,6 +4,7 @@ import backendconfig "github.com/rudderlabs/rudder-server/backend-config"
 
 const (
 	VerifyingObjectStorage       = "Verifying Object Storage"
+	VerifyingObjectStorageDelete = "Verifying delete permissions in object storage"
```

Comment:
Shall it be part of existing step, `Verifying Object Storage`. Current behaviour:
1. Upload file.
2. Download file.

New behaviour: Based on whether the CleanupObjectStorageFilesSetting config is enabled:
1. Upload file.
2. Download file.
3. Delete file.

Reply:
done